app/controllers/api/databases.php

```diff
@@ -3345,25 +3342,44 @@ function updateAttribute(
                         $relation = new Document($relation);
                     }
                     if ($relation instanceof Document) {
-                        $oldDocument = Authorization::skip(fn() => $dbForProject->getDocument(
+                        $relatedDocumentOldVersion = Authorization::skip(fn() => $dbForProject->getDocument(
                             'database_' . $database->getInternalId() . '_collection_' . $relatedCollection->getInternalId(),
                             $relation->getId()
                         ));
 
-                        if ($oldDocument->isEmpty()) {
+                        // If the child document has to be created it will need checking permissions.
+                        if ($relatedDocumentOldVersion->isEmpty()) {
                             $type = Database::PERMISSION_CREATE;
 
                             if (isset($relation['$id']) && $relation['$id'] === 'unique()') {
                                 $relation['$id'] = ID::unique();
                             }
                         } else {
-                            $relation->removeAttribute('$collectionId');
-                            $relation->removeAttribute('$databaseId');
-                            $relation->setAttribute('$collection', $relatedCollection->getId());
                             $type = Database::PERMISSION_UPDATE;
+                            $skipCheckingPermission = true;
+
+                            foreach ($relation as $key => $value) {
+                                //No need to compare values of relations as for each relation we are recursively checking permission.
+                                if ($relatedDocumentOldVersion->getAttribute($key) instanceof Document) {
+                                    continue;
+                                }
+                                //If any of the values are different, we need to check permission.
+                                if ($relatedDocumentOldVersion->getAttribute($key) !== $value) {
+                                    $skipCheckingPermission = false;
+                                    $relation->removeAttribute('$collectionId');
+                                    $relation->removeAttribute('$databaseId');
+                                    $relation->setAttribute('$collection', $relatedCollection->getId());
+                                    break;
+                                }
+                            }
+                        }
+                        if ($skipCheckingPermission) {
+                            Authorization::skip(
+                                fn() => $checkPermissions($relatedCollection, $relation, $relatedDocumentOldVersion, $type)
+                            );
```

Comment:
This skip is slightly concerning...is it needed because looking at the `$checkPermissions()` function, all the db calls already have use skip.